src/test/compile-fail/consts/issue-55878.rs

```diff
@@ -0,0 +1,7 @@
+// normalize-stderr-64bit "18446744073709551615" -> "SIZE"
+// normalize-stderr-32bit "4294967295" -> "SIZE"
+
+// error-pattern: is too big for the current architecture
+fn main() {
+    println!("Size: {}", std::mem::size_of::<[u8; std::u64::MAX as usize]>());
+}
```

Comment:
What is the error here for 32bit vs 64bit, that this cannot be handled by normalization?

Reply:
Missing a span from (at least) one specific target that is present in others:

```
error[E0080]: the type `[u8; SIZE]` is too big for the current architecture	
  --> $SRC_DIR/libcore/mem/mod.rs:LL:COL	
   |	
LL |     intrinsics::size_of::<T>()	
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^	
   | 	
  ::: $DIR/issue-55878.rs:6:26	
   |	
LL |     println!("Size: {}", std::mem::size_of::<[u8; std::u64::MAX as usize]>());	
   |                          ---------------------------------------------------	
```

vs

```
error[E0080]: the type `[u8; SIZE]` is too big for the current architecture	
   | 	
  ::: $DIR/issue-55878.rs:6:26	
   |	
LL |     println!("Size: {}", std::mem::size_of::<[u8; std::u64::MAX as usize]>());	
   |                          ---------------------------------------------------	
```

Reply:
That's extremely odd, isn't it?

Normalization could still handle this -- or you use the two-test-cases approach with `ignore-32bit` and `ignore-64bit`? AFAIK `compile-fail` is on the way out.

Reply:
The divergence is not due to 32bit vs 64bit, it's because the `dist-i586-gnu-i586-i686-musl` `rustc` seems to be packaged slightly different than the other targets, so we miss `core` and `std` information to point spans to. Adding this info was a relatively recent improvement from the last year.

I agree that `compile-fail` should be discouraged, but short of waiting for us to fix the packaging issue I don't see a good solution to this.

Reply:
Oh I see.

Is it possible to ignore musl targets, or so?

Reply:
It is but I have no way of knowing a priori wether this is a `musl`-only discrepancy or not.

Reply:
I'm not super happy about adding more compile-fail tests, but oh well... :)